### PR TITLE
Fix collections.abc imports

### PR DESCRIPTION
**Pull request by antonagestam**
- Abstract base classes are no longer exposed in the top-level collections module in Python 3.9: https://docs.python.org/3.9/whatsnew/3.9.html#removed
- Consolidate docstrings in `util.oset.new`
- Remove unnecessary list call in `util.oset.new`

Attempt to fix these warnings:

```
/app/venv/lib/python3.8/site-packages/spyne/util/oset/new.py:10: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3, and in 3.9 it will stop working
  class oset(collections.MutableSet):
/app/venv/lib/python3.8/site-packages/spyne/protocol/dictdoc/hier.py:29: DeprecationWarning: Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3, and in 3.9 it will stop working
  from collections import defaultdict, Iterable as AbcIterable
```

### CODE REVIEW

**Comment by plq** [comment]
Hello Anton,

I think a minor fix is needed:

```
Python 2.7.17 (default, Dec 16 2019, 02:12:27) 
[GCC 9.2.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> from collections.abc import Iterable as AbcIterable
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: No module named abc
>>> 
```

Thanks!

**Comment by antonagestam** [comment]
@plq Added  a fallback to the old location by catching the `ImportError` :)

**Comment by plq** [comment]
You should have used ``spyne.util.six``

https://github.com/benjaminp/six/commit/0dc54ead62c7fb06edabf1ff22fb71c0c6d958e1

**Comment by antonagestam** [comment]
Gotcha. Didn't find six in dependencies so took for granted it wasn't used. I'll have another go.

**Comment by antonagestam** [comment]
- Updated vendored six to latest version
- Re-applied f1fbfbafca558893c36a0fafc1d4c355175e6f7f
- Realized `spyne.util.oset.new` doesn't need backwards compatibility since `spyne.util.oset.old` handles Python 2.7

**Comment by plq** [comment]
Jenkins, run tests

**Comment by antonagestam** [comment]
@plq Thanks for running the test suite, I've spotted my mistake.

Turns out `old` was an implementation for Python versions <2.7 so can be safely removed. Looking at [the docs](http://spyne.io/docs/2.10/reference/util.html) the `old` and `new` modules under `spyne.util.oset` aren't documented. Would it be okay to flatten the structure and move `spyne/util/oset/new.py` to `spyne/util/oset.py`?

**Comment by antonagestam** [comment]
Removed the legacy implementation of oset and reordered the packages, let me know if you'd want that reverted.

**Comment by plq** [comment]
hello anton,

i don't understand why this patch needs to be so big. please see 7f029a4eac2938eddbc57eff8f05bae8e06c8373
e82ab93b609ed33b69980882d38ea7570d86c025 and 006bffef74113f85493f18c01a3e847f49962b92 and comment whether these are enough to solve the issue with python 3. if so, we can close this pull request. stuff in my fork gets eventually merged so it's as good as it's in the main repo.

best,

**Comment by antonagestam** [comment]
@plq Is it removing the `oset.old` module you're uncomfortable with? I don't understand why you would want to keep it, is support for Python versions *less than* 2.7 a goal?

Other than that I can see you've removed the cleanups I made, of course these don't *need* to happen but was made for good measure. How do you prefer to make cleanups?

**Comment by plq** [comment]
The main reason is that the title says "Fix collections.abc imports". Your patch contains bunch of other things. I'm just trying to do my best reviewing stuff in the limited amount of time I have. So I looked at the title and then the patch size and said "This smells fishy, this patch shouldn't be this complicated". 

If you send your cleanups in another pull request, I'll be happy to look at them.

I also hate touching legacy code for no good reason. Spyne's custom containers differ in subtle ways from Python's standardized ones that come with very similar or even identical names. However, I had another look and in this case you're correct that removing the old oset is harmless. I'll take care of that as well.

Again, sorry if this sounds needlessly bureaucratic but I'm just a lone dev trying to avoid future trouble :)

**Comment by antonagestam** [comment]
Gotcha, no worries, I'll try to keep PRs small and focused.

> I also hate touching legacy code for no good reason.

I can empathize with this sentiment, that's why I tried to limit my cleanups to the files that I had to touch anyway.

Thanks for reviewing and merging!

**Comment by plq** [comment]
Thank you for your understanding. 

FYI: 8c33df3d88460b591e5f4295bbacec04c8fdb50d

Closing this issue. Reopen if not happy.

**Comment by antonagestam** [comment]
@plq What happened with this PR? Did you want me to open a new one with just the `collections.abc` fix? I might have misunderstood you.

**Comment by plq** [comment]
after f029a4 e82ab93 006bffe 8c33df3 there are still legacy imports in the source code?

**Comment by antonagestam** [comment]
@plq What do you mean? It's quite confusing when you reference "loose" commits (are those authored by me but marked as committed by you?).

I'm not seeing any of those commits, either my own or yours on master which is why I'm asking. It's really hard to follow along in your process.

**Comment by plq** [comment]
I'm sorry about the confusion, they are merged to master now.
